docs/source/fetching/persisted-queries.mdx

```diff
@@ -0,0 +1,160 @@
+
+The persisted queries feature requires operations to be preregistered in a **persisted query list** (**PQL**). 
+This allows the PQL to act as a safelist of trusted operations made by your first-party apps. As such, persisted queries is a security feature as much as a performance one.
+
+With APQs, if the server can't find the operation ID the client provides, the server returns an error indicating that it needs the full operation string. If the Apollo iOS client receives this error, it automatically retries the operation with the full operation string.
+
+If you _only_ want to improve request latency and bandwidth usage, APQ addresses your use case. If you _also_ want to secure your supergraph with operation safelisting, you should preregister trusted operations in a PQL.
+
+For more details on differences between persisted queries and APQ, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries#differences-from-automatic-persisted).
+
+### Requirements
+
+Persisted queries is currently in [preview](/resources/product-launch-stages#preview) and has the following requirements:
+- Apollo iOS (v1.3.1+)
+- [Apollo Router](/router) (v1.25.0+)
+- [GraphOS Enterprise plan](/graphos/enterprise/)
+
+You can use APQ with the following versions of Apollo iOS, Apollo Router, and Apollo Server:
+- Apollo iOS (v1.0.0+)
+- [Apollo Router](/router) (v0.1.0+)
+- [Apollo Server](/apollo-server/) (v1.0.0+)
+
+## Implementation steps
+
+Both persisted queries and APQ require you to configure code generation and how your client makes requests. The rest of this article details these configurations.
+
+Persisted queries also requires you to create and link a PQL and configure your router. For more information on the other configuration aspects of persisted queries, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries).
+
+### 1. Configure code generation
+
+The code generation requirements for persisted queries and APQ are the same: your operations need a SHA256 hash identifier included in the generated operation objects. To configure code generation accordingly, make the following changes in your `ApolloCodegenConfiguration` JSON file:
+
+1. In the `output` section of your configuration, set the [`operationManifest`](./../code-generation/codegen-configuration#operation-manifest) with:
+- a `path` to output the manifest file
+- the `version`: either `persistedQueries` for persisted queries or `legacyAPQs` for APQs
+
+```json title="apollo-codegen-config.json"
+"output": {
+    "operationManifest" : {
+        "path" : "./generated/operationIdentifiers.json",
+        "version" : "persistedQueries"
+    }
+}
+```
+
+2. In the `options` section of your configuration, set the `operationDocumentFormat` array to `definition`, `operationId`, or both `definition` and `operationId`. This determines how to generate the operation documents for your generated operations.
```

Comment:
It needs to be stated here that for `APQs` to work, you **MUST** include both the `definition` and `operationId`. For `persistedQueries`, you only need the `operationId`, the `definition` is superflous, since the definition is already persisted.